internal/subscription/core.go

```diff
@@ -344,6 +345,79 @@ func createDelayTopics(ctx context.Context, m *Model, topicCore topic.ICore, top
 	return nil
 }
 
+// RescaleSubTopics - Get all the subs and rescale all the Retry/Delay/DLQ topics
+func (c *Core) RescaleSubTopics(ctx context.Context, topicModel *topic.Model) error {
+	projectList, err := c.projectCore.GetAllProjects(ctx)
+	if err != nil {
+		return err
+	}
+	var completeSubList []*Model
+
+	for _, projectKey := range projectList {
+		subList, err := c.List(ctx, Prefix+project.FetchProjectID(projectKey))
+		if err != nil {
+			return err
+		}
+		completeSubList = append(completeSubList, subList...)
+	}
+
+	for _, m := range completeSubList {
+		if m.ExtractedTopicName != topicModel.ExtractedTopicName {
+			logger.Ctx(ctx).Error("Subscription not part of the topic. Sub: ", m.Name)
+			continue
+		}
+		retryModel := &topic.Model{
+			Name:               m.GetRetryTopic(),
+			ExtractedTopicName: m.ExtractedSubscriptionName + topic.RetryTopicSuffix,
+			ExtractedProjectID: m.ExtractedTopicProjectID,
+			NumPartitions:      topicModel.NumPartitions,
+		}
+		retryTopicError := c.topicCore.UpdateTopic(ctx, retryModel)
+		if retryTopicError != nil {
+			logger.Ctx(ctx).Error(
+				"Error in executing Retry Topic Rescaling: ",
+				retryTopicError.Error(),
+				" TopicName: ",
+				retryModel.Name)
```

Comment:
if there is any error in updating partitions, do we want to go ahead with other topic modifications..?
Is it intentional

Reply:
Sometime the possible Topic on subs are not present, for ex the retry topic doesn't exists and retry happens via original topic. In that case this would throw and error and we still want to continue.

Reply:
Okay, when retry topics are not created how are the retries carried out?

Reply:
Retry topics are always created. If not the topic create request fails.